python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -633,6 +645,118 @@ def is_successful(self) -> bool:
+        materialized by dbt.
+        """
+        adapter = check.not_none(self.adapter)
+
+        dbt_resource_props = self._get_dbt_resource_props_from_event(event)
+        is_view = dbt_resource_props["config"]["materialized"] == "view"
+
+        # Avoid counting rows for views, since they may include complex SQL queries
+        # that are costly to execute. We can revisit this in the future if there is
+        # a demand for it.
+        if is_view:
+            return event
+
+        # If the adapter is DuckDB, we need to wait for the dbt CLI process to complete
+        # so that the DuckDB lock is released. This is because DuckDB does not allow for
+        # opening multiple connections to the same database when a write connection, such
+        # as the one dbt uses, is open.
+        try:
+            from dbt.adapters.duckdb import DuckDBAdapter
+
+            if isinstance(adapter, DuckDBAdapter):
+                self._dbt_run_thread.result()
+        except ImportError:
+            pass
+
+        unique_id = dbt_resource_props["unique_id"]
+        table_str = f"{dbt_resource_props['database']}.{dbt_resource_props['schema']}.{dbt_resource_props['name']}"
+
+        with adapter.connection_named(f"row_count_{unique_id}"):
+            query_result = adapter.execute(
+                f"""
+                    SELECT
+                    count(*) as row_count
+                    FROM
+                    {table_str}
+                """,
+                fetch=True,
+            )
+        row_count = query_result[1][0]["row_count"]
+        additional_metadata = {**TableMetadataSet(row_count=row_count)}
+
+        if isinstance(event, Output):
+            return event.with_metadata(metadata={**event.metadata, **additional_metadata})
+        else:
+            return event._replace(metadata={**event.metadata, **additional_metadata})
+
+    def _stream_dbt_events_and_enqueue_postprocessing(
+        self,
+        output_events_and_futures: List[Union[Future, DbtDagsterEventType]],
+        executor: ThreadPoolExecutor,
+    ) -> None:
+        """Task which streams dbt events and either directly places them in
+        the output_events list to be emitted to user code, or enqueues post-processing tasks
+        where needed.
+        """
+        for event in self._stream_asset_events():
+            # For any materialization or output event, we run postprocessing steps
+            # to attach additional metadata to the event.
+            if self.should_fetch_row_counts and isinstance(event, (AssetMaterialization, Output)):
+                output_events_and_futures.append(
+                    executor.submit(
+                        self._attach_post_materialization_metadata,
+                        event,
+                    )
+                )
+            else:
+                output_events_and_futures.append(event)
+
+    @experimental
+    def fetch_row_counts(
```

Comment:
Maybe remove the plural?

Also, I wonder if this method is ill-named.

The current method name insinuates that the fetch is happening after the method is called, which is not true. It is setting an attribute for `.stream()` to read from.

Some suggestions:

- `with_row_count`
- `with_row_count_retrieval`
- `with_row_count_fetch`
- `enable_row_count_retrieval`
- `enable_fetch_row_count`

Reply:
Yeah, I like `enable_fetch_row_count`

Reply:
Maybe worth chatting about in standup, but the plural sounds right to me. Multiple row counts are being fetched.

Reply:
> insinuates that the fetch is happening after the method is called, which is not true.

Should we definitely not do the fetch when the method is called?

Reply:
To do so we'd need to stream/process the dbt event stream in some form, which we typically don't do until the user calls `stream()`.